src/api/app/models/workflow/step/branch_package_step.rb

```diff
@@ -120,6 +122,19 @@ def branch_request_file_gitlab
           project: { http_url: @scm_extractor_payload[:http_url] },
           object_attributes: { source: { default_branch: @scm_extractor_payload[:commit_sha] } } }.to_json
       end
+
+      def set_or_update_subscriptions(package)
+        ['Event::BuildFail', 'Event::BuildSuccess'].each do |build_event|
+          subscription = EventSubscription.first_or_create!(eventtype: build_event,
+                                                            receiver_role: 'reader', # We pass a valid value, but we don't need this.
+                                                            user: User.session.login,
```

Comment:
Please don't forget to use `@token.user` here unless you plan to do it in another commit/PR.

And do the same in the `branch` method.